File: frontend/src/types/market.ts
export interface TopGainerItem {
  symbol: string;
  identifier?: string;
  industry?: string;
  lastPrice: number;
  previousClose: number;
  change: number;
  pChange: number;
  totalTradedVolume: number;
  lastUpdateTime: string;
}

export interface TopGainersResponse {
  index: string;
  min_pct_gain: number;
  count: number;
  gainers: TopGainerItem[];
}

File: backend/app/schemas/market.py
from pydantic import BaseModel
from typing import List, Optional

class TopGainerItem(BaseModel):
    symbol: str
    identifier: Optional[str] = None
    industry: Optional[str] = None
    lastPrice: float
    previousClose: float
    change: float
    pChange: float
    totalTradedVolume: int
    lastUpdateTime: str
    smaValue: Optional[float] = None
    smaDistance: Optional[float] = None

class TopGainersResponse(BaseModel):
    index: str
    min_pct_gain: float
    count: int
    gainers: List[TopGainerItem]
